File: app/src/main/java/com/polok/eubmanagement/presentation/notice/noticelist/NoticeListAdapter.java
package com.polok.eubmanagement.presentation.notice.noticelist;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.polok.eubmanagement.databinding.ItemNoticeListBinding;
import com.polok.eubmanagement.model.NoticeData;
import java.util.List;

public class NoticeListAdapter extends RecyclerView.Adapter<NoticeListAdapter.NoticeViewHolder> {
    private final List<NoticeData> noticeList;
    private final View.OnClickListener onClickListener;
    private int adapterPosition;
    public NoticeListAdapter(List<NoticeData> noticeList, View.OnClickListener onClickListener) {
        this.noticeList = noticeList;
        this.onClickListener = onClickListener;
    }

    @NonNull
    @Override
    public NoticeListAdapter.NoticeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new NoticeViewHolder(
                ItemNoticeListBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false)
        );
    }

    @Override
    public void onBindViewHolder(@NonNull NoticeListAdapter.NoticeViewHolder holder, int position) {
        holder.bind(noticeList.get(position));
    }

    @Override
    public int getItemCount() {
        return noticeList.size();
    }

    public class NoticeViewHolder extends RecyclerView.ViewHolder {
        ItemNoticeListBinding binding;
        public NoticeViewHolder(@NonNull ItemNoticeListBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
        public void bind(NoticeData noticeData) {
            binding.noticeTitle.setText(noticeData.getNotNullText(noticeData.getTitle()));
            binding.noticeDetails.setText(noticeData.getNotNullText(noticeData.getDetails()));
            binding.createdAt.setText(String.format("Created At: %s", noticeData.getNotNullText(noticeData.getCreatedAt())));

            if (onClickListener != null) {
                itemView.setOnClickListener(v -> {
                    adapterPosition = getAdapterPosition();
                    onClickListener.onClick(itemView);
                });
            } else binding.getRoot().setClickable(false);
        }
    }

    public int getAdapterPosition() {return adapterPosition;}
}

File: app/src/main/java/com/polok/eubmanagement/presentation/notice/noticelist/NoticeListAdapter.kt
package com.polok.eubmanagement.presentation.notice.noticelist

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.res.ResourcesCompat
import androidx.recyclerview.widget.RecyclerView
import com.polok.eubmanagement.R
import com.polok.eubmanagement.databinding.ItemNoticeListBinding
import com.polok.eubmanagement.model.NoticeData
import com.polok.eubmanagement.presentation.notice.noticelist.NoticeListAdapter.NoticeViewHolder
import com.polok.eubmanagement.util.makeVisible

class NoticeListAdapter(
    private val onClickListener: ((NoticeData?) -> Unit)?
) : RecyclerView.Adapter<NoticeViewHolder>() {
    var isAdmin: Boolean? = null
    private var noticeList: List<NoticeData?>? = null

    fun submitList(noticeList: List<NoticeData?>?) {
        this.noticeList = noticeList
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoticeViewHolder {
        return NoticeViewHolder(
            ItemNoticeListBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: NoticeViewHolder, position: Int) {
        holder.bind(noticeList?.get(position))
    }

    override fun getItemCount(): Int = noticeList?.size ?: 0

    inner class NoticeViewHolder(
        private val binding: ItemNoticeListBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(noticeData: NoticeData?) {
            binding.noticeTitle.text = noticeData?.title ?: ""
            binding.noticeDetails.text = noticeData?.details ?: ""
            binding.createdAt.text = String.format("Created At: %s", noticeData?.createdAt ?: "")

            if (onClickListener != null) {
                itemView.setOnClickListener {
                    onClickListener.invoke(noticeData)
                }
            } else {
                binding.getRoot().isClickable = false
                binding.root.setBackgroundResource(R.drawable.edit_text_rounded_primary_border_bg)
            }

            if (isAdmin == true) {
                binding.updateDelete.root.makeVisible()
            }
        }
    }
}
